ipfx/ephys_data_set.py

```diff
@@ -197,17 +195,41 @@ def get_recording_date(self):
 
     def get_sweep_data(self, sweep_number):
         """
-        Return the data of sweep_number as dict. The dict has the format:
+        Read sweep data from the nwb file
+        Substitute trailing zeros in the response for np.nan
+        because trailing zeros occur after the end of recording
+
+        Parameters
+        ----------
+        sweep_number
+
+        Returns
+        -------
+        dict in the format:
 
         {
             'stimulus': np.ndarray,
             'response': np.ndarray,
             'stimulus_unit': string,
             'sampling_rate': float
         }
-
         """
-        raise NotImplementedError
+
+        sweep_data = self.nwb_data.get_sweep_data(sweep_number)
```

Comment:
If you are relying on the `self.nwb_data` attribute then you should create something like to indicate that subclasses should specify `nwb_data`:
```
@property
def nwb_data(self):
    raise NotImplementedError
```

Reply:
Thanks @njmei. I have implemented the property as you suggested